File: src/getSettings.cpp
#include "getSettings.h"

CSconesSettings getSettings(List userSettings) {

  CSconesSettings settings;
  settings = CSconesSettings();

  // unsigned int
  if (userSettings.containsElementNamed("folds"))
    settings.folds = userSettings["folds"];

  // bool, VectorXd, VectorXd
  if (userSettings.containsElementNamed("autoParameters") &
      userSettings.containsElementNamed("lambdas") &
      userSettings.containsElementNamed("etas")){
    settings.autoParameters = userSettings["autoParameters"];
    settings.lambdas = userSettings["lambdas"];
    settings.etas = userSettings["etas"];
  }

  // unsigned int
  if (userSettings.containsElementNamed("test_statistic"))
    settings.test_statistic = userSettings["test_statistic"];

  // unsigned int
  if (userSettings.containsElementNamed("nParameters"))
    settings.nParameters = userSettings["nParameters"];

  // unsigned int
  if (userSettings.containsElementNamed("gridsearch_depth"))
    settings.gridsearch_depth = userSettings["gridsearch_depth"];

  // unsigned int
  if (userSettings.containsElementNamed("selection_criterion"))
    settings.selection_criterion = userSettings["selection_criterion"];

  // double
  if (userSettings.containsElementNamed("seed"))
    settings.seed = userSettings["seed"];

  // double
  if (userSettings.containsElementNamed("selection_ratio"))
    settings.selection_ratio = userSettings["selection_ratio"];

  // bool
  if (userSettings.containsElementNamed("dump_intermediate_results"))
    settings.dump_intermediate_results = userSettings["dump_intermediate_results"];

  // bool
  if (userSettings.containsElementNamed("evaluateObjective"))
    settings.evaluateObjective = userSettings["evaluateObjective"];

  return settings;

}

File: R/getSettings.R
getGinSettings <- function(...){

  settings = list(...)

  # unsigned int
  if (! "nParameters" %in% settings)
    settings["nParameters"] = 10;

  # unsigned int
  if (! "folds" %in% settings)
    settings["folds"] = 10

  # bool, VectorXd, VectorXd
  if (! "autoParameters" %in% settings & "lambdas" %in% settings & "etas" %in% settings) {
    settings["autoParameters"] = TRUE
    settings["lambdas"] = rep(0, settings["nParameters"])
    settings["etas"] = rep(0, settings["nParameters"])
  }

  # unsigned int
  if (! "test_statistic" %in% settings)
    settings["test_statistic"] = 0

  # unsigned int
  if (! "gridsearch_depth" %in% settings)
    settings["gridsearch_depth"] = 1

  # unsigned int
  if (! "selection_criterion" %in% settings)
    settings["selection_criterion"] = 1

  # double
  if (! "seed" %in% settings)
    settings["seed"] = 0

  # double
  if (! "selection_ratio" %in% settings)
    settings["selection_ratio"] = 0.8

  # bool
  if (! "dump_intermediate_results" %in% settings)
    settings["dump_intermediate_results"] = TRUE

  # bool
  if (! "evaluateObjective" %in% settings)
    settings["evaluateObjective"] = FALSE

  return(settings);
}
